test/plugin/test_out_detect_exceptions.rb

```diff
@@ -98,6 +99,54 @@ def test_exception_detection
     assert_equal(make_logs(t, *messages), d.events)
   end
 
+  def test_ignore_exception_nested_in_json
+    cfg = 'languages python'
+    d = create_driver(cfg)
+    t = Time.now.to_i
+
+    # There is a nested exception within the body, we should ignore those!
+    lines = <<END
```

Comment:
Trying to understand the use case here: Seems to me this issue only exists if the customer does not enable `in_tail` plugin with `json` format, right? Because `in_tail` plugin is an input plugin. It comes before `fluent-plugin-detect-exceptions`, an output plugin. Shouldn't the fix be from the customer side's configuration then? Aka, if the test mocks exactly why they ran into the issue in the first place (might worth confirming with them on what logs they send), they should fix their config to enable `format json` in their config. As mentioned in the other comment, I'm concerned of a potentially breaking change introduced by this fix. Also the rules for most of the other languages do not require the keywords to be at the beginning of the lines either.